### PR TITLE
[ADDED] Account default permissions

### PR DESCRIPTION
**Pull request by matthiashanel**
Adding default permissions to account (config and jwt).
This simplifies config of the common user. 
This happens in the authorization block already.

This change corresponds to https://github.com/nats-io/jwt/pull/75

### CODE REVIEW

File: server/accounts.go [hunk 1]
```diff
@@ -2087,6 +2088,23 @@ func buildInternalNkeyUser(uc *jwt.UserClaims, acc *Account) *NkeyUser {
...
+	nu := &NkeyUser{Nkey: uc.Subject, Account: acc}
+	if uc.IssuerAccount != "" {
+		nu.SigningKey = uc.Issuer
+	}
+
+	// Now check for permissions.
+	var p = buildPermissionsFromJwt(&uc.Permissions)
+	if p == nil && acc.default_perms != nil {
+		json, _ := json.Marshal(*acc.default_perms)
+		log.Printf("apply default_perms %+v to %s\n", string(json), uc.Subject)
```
Comment (kozlovic) [inline server/accounts.go:2105]: debugging left over I assume

File: server/accounts.go [hunk 2]
```diff
@@ -2087,6 +2088,23 @@ func buildInternalNkeyUser(uc *jwt.UserClaims, acc *Account) *NkeyUser {
...
+		p = acc.default_perms
```
Comment (kozlovic) [inline server/accounts.go:2106]: We have to ask ourselves: would there be a case where we alter this user's permissions after that. If so, that would change the account's default user. It exists a `Permissions.clone()` that makes a copy of permissions. If need be, we may need to use that here.

File: server/opts.go [hunk 3]
```diff
@@ -2199,6 +2211,23 @@ func parseImportStreamOrService(v interface{}, errors, warnings *[]error) (*impo
...
 }
 
+// Apply permission defaults to users/nkeyuser that don't have their own.
+func applyDefaultPermissions(users []*User, nkeys []*NkeyUser, defaultP *Permissions) {
+	if defaultP == nil {
+		return
+	}
+	for _, user := range users {
+		if user.Permissions == nil {
+			user.Permissions = defaultP
```
Comment (kozlovic) [inline server/opts.go:2221]: Here and down below, same comment I made about possibly use `.clone()`

File: server/accounts.go [hunk 4]
```diff
@@ -70,6 +72,7 @@ type Account struct {
 	siReply       []byte  // service reply prefix, will form wildcard subscription.
 	siReplyClient *client
 	prand         *rand.Rand
+	default_perms *Permissions
```
Comment (wallyqs) [inline server/accounts.go:75]:
```suggestion
	defaultPerms *Permissions
```

File: server/accounts.go (hunk 2 above)
Comment (matthiashanel) [reply server/accounts.go:2106]: I went back and forth on this. Maybe it's safest to clone them

File: server/accounts.go (hunk 4 above)
Comment (matthiashanel) [reply server/accounts.go:75]: will change

File: server/accounts.go (hunk 2 above)
Comment (matthiashanel) [reply server/accounts.go:2106]: @kozlovic the reason for not calling clone was that auth.defaultPermissions was not cloned either. Change it for both I assume?

Comment (kozlovic) [reply server/accounts.go:2106]: I would clone, unless we are sure they will never be altered (I mean at the user-level, consequently altering the default values).

Comment (kozlovic) [reply server/accounts.go:2106]: A precision to that: if we start to have lots and lots of users, and we are sure that if there are default perms we don't alter them, then we could save mem allocs. So I guess it is more a question of ensuring that if a user uses default perms, then we never ever modify them. Or if we have to, then that's when we do a clone.

Comment (matthiashanel) [reply server/accounts.go]: @kozlovic  I noticed that when user are copied from options to the server, their permissions are cloned. This is why I don't call clone in that code. However when generating an nkey user on the fly I did add clone to arrive at the same data structure as when configured.

File: server/config_check_test.go [hunk 5]
```diff
@@ -750,8 +750,8 @@ func TestConfigCheck(t *testing.T) {
                 http_port = 8222
 				`,
 			err:       errors.New(`Duplicate user "foo" detected`),
-			errorLine: 6,
-			errorPos:  21,
+			errorLine: 5,
```
Comment (kozlovic) [inline server/config_check_test.go:753]: Revert this change and see below why...

File: server/opts.go [hunk 6]
```diff
@@ -1788,6 +1788,10 @@ func parseAccounts(v interface{}, opts *Options, errors *[]error, warnings *[]er
 				*errors = append(*errors, err)
 				continue
 			}
+			var (
+				users   []*User
+				nkeyUsr []*NkeyUser
+			)
```
Comment (kozlovic) [inline server/opts.go:1794]: Declare `usersTk token` here.

File: server/opts.go [hunk 7]
```diff
@@ -1825,32 +1829,19 @@ func parseAccounts(v interface{}, opts *Options, errors *[]error, warnings *[]er
 						continue
 					}
 				case "users":
-					nkeys, users, err := parseUsers(mv, opts, errors, warnings)
+					var err error
```
Comment (kozlovic) [inline server/opts.go:1832]: Add `usersTk = tk` here.

File: server/opts.go [hunk 8]
```diff
@@ -1863,6 +1854,27 @@ func parseAccounts(v interface{}, opts *Options, errors *[]error, warnings *[]er
 					}
 				}
 			}
+			applyDefaultPermissions(users, nkeyUsr, acc.defaultPerms)
+			for _, u := range nkeyUsr {
+				if _, ok := uorn[u.Nkey]; ok {
+					err := &configErr{tk, fmt.Sprintf("Duplicate nkey %q detected", u.Nkey)}
```
Comment (kozlovic) [inline server/opts.go:1860]: Replace `tk` with `usersTk` here.

File: server/opts.go [hunk 9]
```diff
@@ -1863,6 +1854,27 @@ func parseAccounts(v interface{}, opts *Options, errors *[]error, warnings *[]er
...
+					*errors = append(*errors, err)
+					continue
+				}
+				uorn[u.Nkey] = struct{}{}
+				u.Account = acc
+			}
+			opts.Nkeys = append(opts.Nkeys, nkeyUsr...)
+			for _, u := range users {
+				if _, ok := uorn[u.Username]; ok {
+					err := &configErr{tk, fmt.Sprintf("Duplicate user %q detected", u.Username)}
```
Comment (kozlovic) [inline server/opts.go:1870]: Replace `tk` with `usersTk` here.

File: server/config_check_test.go (hunk 5 above)
Comment (matthiashanel) [reply server/config_check_test.go]: makes perfect sene
